lib/src/controls/form/number_box.dart

```diff
@@ -415,7 +425,21 @@ class NumberBoxState extends State<NumberBox> {
     _updateValue();
   }
 
+  int _constrainValue(int value) {
+    if (widget.min != null && widget.min! > value) {
+      return widget.min!;
+    }
+
+    if (widget.max != null && widget.max! < value) {
+      return widget.max!;
+    }
+
+    return value;
+  }
```

Comment:
I would make use of the dart's math lib to shortcut this code

```suggestion
  int _constrainValue(int value) {
    return math.min(
      widget.min ?? double.infinity,
      math.max(
        widget.max ?? double.infinity,
        value,
      ),
    );
  }
```

Reply:
Not sure that's more readable, but shorter nonetheless.

Reply:
@bdlukaa I agree with @ricardoboss, his code looks simpler and more readable 😅